src/core/tools/common/simple_logger.h

```diff
@@ -33,6 +33,8 @@
 namespace dsn {
 namespace tools {
 
+void set_log_prefixed_message_func(std::function<const std::string()> func);
```

Comment:
```suggestion
void set_log_prefixed_message_func(std::function<std::string()> func);
```

Place this function in `logging_provider.h`. Because `log_prefixed_message_func` is not limited to `simple_logger`, it's shared with other implementations (though we do not have now).